include/core/lethe_grid_tools.h

```diff
@@ -316,6 +316,29 @@ namespace LetheGridTools
   find_point_triangle_distance(const std::vector<Point<dim>> &triangle_vertices,
                                const Point<dim>              &point);
 
+
+  /**
+   * @brief Calculates the intersection points between a line and a sphere. The
+   * full calculation is taken from  Geometric Tools for Computer Graphics,
+   * Eberly 2003 Chapter 11.3.2 - LINEAR COMPONENTS AND A SPHERE. The entire
+   * reference is available at:
+   * https://www.sciencedirect.com/science/article/pii/B978155860594750014X
+   * 
+   * @param line_start The starting point of the line
+   * @param line_direction The direction of the line
+   * @param sphere_center The center of the sphere
+   * @param sphere_radius The radius of the sphere
+   * @param intersection_points A vector to store the intersection points
```

Comment:
```suggestion
   * @param[in] line_start The starting point of the line
   * @param[in] line_direction The direction of the line
   * @param[in] sphere_center The center of the sphere
   * @param[in] sphere_radius The radius of the sphere
   * @param[out] intersection_points A vector to store the intersection points
```

Maybe mention if the vector need to be pre-allocated or not during.